From 5fe11f6db79f46661fae7c052bce5c6ed1f9cc34 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <Xinyu.Chen@freescale.com>
Date: Tue, 17 Feb 2009 15:19:56 +0800
Subject: [PATCH] ENGR00107568 MX51: Add support to CLAA WVGA LCD panel

Add support to CLAA WVGA panel, without lcd backlight control

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 arch/arm/mach-mx51/mx51_3stack.c      |   11 +++++++++++
 arch/arm/mach-mx51/mx51_3stack_gpio.c |   10 ++++++++++
 2 files changed, 21 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx51/mx51_3stack.c b/arch/arm/mach-mx51/mx51_3stack.c
index d6266e0..693e5e4 100644
--- a/arch/arm/mach-mx51/mx51_3stack.c
+++ b/arch/arm/mach-mx51/mx51_3stack.c
@@ -283,6 +283,16 @@ static struct platform_device mxc_lcd_device = {
 		},
 };
 
+static struct mxc_lcd_platform_data lcd_wvga_data;
+
+static struct platform_device lcd_wvga_device = {
+	.name = "lcd_claa",
+	.dev = {
+		.release = mxc_nop_release,
+		.platform_data = &lcd_wvga_data,
+		},
+};
+
 extern void gpio_lcd_active(void);
 
 static void mxc_init_fb(void)
@@ -293,6 +303,7 @@ static void mxc_init_fb(void)
 		lcd_data.reset = lcd_reset_to2;
 
 	(void)platform_device_register(&mxc_lcd_device);
+	(void)platform_device_register(&lcd_wvga_device);
 
 	(void)platform_device_register(&mxc_fb_device[0]);
 	(void)platform_device_register(&mxc_fb_device[1]);
diff --git a/arch/arm/mach-mx51/mx51_3stack_gpio.c b/arch/arm/mach-mx51/mx51_3stack_gpio.c
index b0f5a48..f996009 100644
--- a/arch/arm/mach-mx51/mx51_3stack_gpio.c
+++ b/arch/arm/mach-mx51/mx51_3stack_gpio.c
@@ -797,8 +797,10 @@ EXPORT_SYMBOL(sdhc_write_protect);
 void gpio_lcd_active(void)
 {
 	if (cpu_is_mx51_rev(CHIP_REV_2_0) > 0) {
+#ifndef CONFIG_FB_MXC_CLAA_WVGA_SYNC_PANEL
 		mxc_request_iomux(MX51_PIN_DI1_D1_CS, IOMUX_CONFIG_ALT4);
 		mxc_set_gpio_direction(MX51_PIN_DI1_D1_CS, 0);
+#endif
 		mxc_request_iomux(MX51_PIN_DI1_D0_CS, IOMUX_CONFIG_ALT1);
 		mxc_request_iomux(MX51_PIN_DI1_PIN11, IOMUX_CONFIG_ALT1);
 		mxc_request_iomux(MX51_PIN_DI1_PIN12, IOMUX_CONFIG_ALT1);
@@ -808,6 +810,14 @@ void gpio_lcd_active(void)
 		mxc_request_iomux(MX51_PIN_DI_GP2, IOMUX_CONFIG_ALT0);
 		mxc_request_iomux(MX51_PIN_DI_GP3, IOMUX_CONFIG_ALT0);
 	}
+#ifdef CONFIG_FB_MXC_CLAA_WVGA_SYNC_PANEL
+	mxc_request_iomux(MX51_PIN_DI1_D1_CS, IOMUX_CONFIG_ALT4);
+	mxc_set_gpio_direction(MX51_PIN_DI1_D1_CS, 0);
+	mxc_set_gpio_dataout(MX51_PIN_DI1_D1_CS, 1);
+	mxc_request_iomux(MX51_PIN_DISPB2_SER_DIO, IOMUX_CONFIG_ALT4);
+	mxc_set_gpio_direction(MX51_PIN_DISPB2_SER_DIO, 0);
+	mxc_set_gpio_dataout(MX51_PIN_DISPB2_SER_DIO, 0);
+#endif
 }
 
 /*!
-- 
1.5.4.4

